metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -366,3 +377,128 @@ def _prepend_platform_instance(
         return [BrowsePathEntryClass(id=urn, urn=urn)] + entries
 
     return entries
+
+
+def auto_incremental_lineage(
+    graph: Optional[DataHubGraph],
+    incremental_lineage: bool,
+    include_column_level_lineage: bool,
```

Comment:
this is kinda confusingly named - maybe we should call it "source_produces_column_lineage" or something

the other possibility is that we mix the two - if the UpstreamLineageClass has fineGrainedLineage, then we use read-modify-write, otherwise we do patch. we'd need to test that patch doesn't delete fineGrainedLineage if it was previously present

Anyways - not a blocker for this PR, but I do want to revisit this